testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetGroupIT.groovy

```diff
@@ -0,0 +1,274 @@
+import static org.sonatype.nexus.testsuite.repository.FormatClientSupport.bytes
+import static org.sonatype.nexus.testsuite.repository.FormatClientSupport.status
+import static org.sonatype.tests.http.server.fluent.Behaviours.file
+
+/**
+ * Test NuGet group repository behaviour
+ */
+@ExamReactorStrategy(PerClass.class)
+class NugetGroupIT
+    extends NugetProxyITSupport
+{
+
+  public static final String PROXY_REPO_NAME = 'test-nuget-proxy'
+
+  public static final String HOSTED_REPO_NAME = 'test-nuget-hosted'
+
+  public static final String GROUP_REPO_HOSTED_FIRST_NAME = 'test-nuget-group-hosted-first'
+
+  public static final String GROUP_REPO_PROXY_FIRST_NAME = 'test-nuget-group-proxy-first'
+
+  public static final String GROUP_REPO_HOSTED_ONLY_NAME = 'test-nuget-group-hosted-only'
+
+  public static final String GROUP_REPO_PROXY_ONLY_NAME = 'test-nuget-group-proxy-only'
+
+  NugetClient proxy
+
+  NugetClient hosted
+
+  NugetClient proxyFirstGroup
+
+  NugetClient hostedFirstGroup
+
+  NugetClient proxyOnlyGroup
+
+  NugetClient hostedOnlyGroup
+
+  static final Map<String, String> SONATYPE_COMPONENT = [
+      id     : 'SONATYPE.TEST',
+      version: '1.0',
+      nupkg  : 'SONATYPE.TEST.1.0.nupkg'
+  ].asImmutable()
+
+  @Before
+  public void createRepositories() {
+    //proxy repo
+    configureProxyResources()
+    final String remoteStorageUrl = proxyServer.getUrl().toExternalForm() + "/" + REMOTE_NUGET_REPO_PATH
+    final Repository proxyRepo = createRepository(proxyConfig(PROXY_REPO_NAME, remoteStorageUrl))
+    proxy = nugetClient(proxyRepo)
+
+    //hosted repo
+    final Repository hostedRepo = createRepository(hostedConfig(HOSTED_REPO_NAME))
+    hosted = nugetClient(hostedRepo)
+
+    //group repos
+    final Repository groupRepoProxyFirst =
+        createRepository(groupConfig(GROUP_REPO_PROXY_FIRST_NAME, PROXY_REPO_NAME, HOSTED_REPO_NAME))
+    proxyFirstGroup = nugetClient(groupRepoProxyFirst)
+    final Repository groupRepoHostedFirst =
+        createRepository(groupConfig(GROUP_REPO_HOSTED_FIRST_NAME, HOSTED_REPO_NAME, PROXY_REPO_NAME))
+    hostedFirstGroup = nugetClient(groupRepoHostedFirst)
+    final Repository groupRepoProxyOnly =
+        createRepository(groupConfig(GROUP_REPO_PROXY_ONLY_NAME, PROXY_REPO_NAME))
+    proxyOnlyGroup = nugetClient(groupRepoProxyOnly)
+    final Repository groupRepoHostedOnly =
+        createRepository(groupConfig(GROUP_REPO_HOSTED_ONLY_NAME, HOSTED_REPO_NAME, PROXY_REPO_NAME))
+    hostedOnlyGroup = nugetClient(groupRepoHostedOnly)
+  }
+
+  @Test
+  void 'Repository metadata is available from the group'() {
+    final String repositoryMetadata = proxyOnlyGroup.getRepositoryMetadata()
+    assertThat(repositoryMetadata, is(notNullValue()))
+    assertThat(repositoryMetadata, containsString('<Schema Namespace="NuGetGallery"'))
+  }
+
+  @Test
+  void 'Initially the group feed contains no data'() {
+    assertThat(parseFeedXml(proxyOnlyGroup.vsSearchFeedXml()).size(), is(0))
```

Comment:
similar, but my personal preference is to keep this part consistent with Java testing